runtime/integration-tests/src/generic/cases/lp/mod.rs

```diff
@@ -0,0 +1,1956 @@
+#[allow(non_camel_case_types)]
+pub struct DAI;
+impl CurrencyInfo for DAI {
+	fn custom(&self) -> CustomMetadata {
+		CustomMetadata {
+			pool_currency: true,
+			transferability: CrossChainTransferability::LiquidityPools,
+			permissioned: false,
+			mintable: false,
+			local_representation: None,
+		}
+	}
+
+	fn symbol(&self) -> &'static str {
+		"DAI"
+	}
+
+	fn id(&self) -> CurrencyId {
+		CurrencyId::ForeignAsset(100_002)
+	}
+
+	fn ed(&self) -> Balance {
+		100_000_000_000_000
+	}
+
+	fn decimals(&self) -> u32 {
+		18
+	}
+}
+
+#[allow(non_camel_case_types)]
+pub struct FRAX;
+impl CurrencyInfo for FRAX {
+	fn custom(&self) -> CustomMetadata {
+		CustomMetadata {
+			pool_currency: true,
+			transferability: CrossChainTransferability::LiquidityPools,
+			permissioned: false,
+			mintable: false,
+			local_representation: None,
+		}
+	}
+
+	fn symbol(&self) -> &'static str {
+		"FRAX"
+	}
+
+	fn id(&self) -> CurrencyId {
+		CurrencyId::ForeignAsset(100_003)
+	}
+
+	fn ed(&self) -> Balance {
+		100_000_000_000_000
+	}
+
+	fn decimals(&self) -> u32 {
+		18
+	}
+}
+
+#[allow(non_camel_case_types)]
+pub struct LocalUSDC;
+impl CurrencyInfo for LocalUSDC {
+	fn custom(&self) -> CustomMetadata {
+		CustomMetadata {
+			pool_currency: true,
+			transferability: CrossChainTransferability::None,
+			permissioned: false,
+			mintable: false,
+			local_representation: None,
+		}
+	}
+
+	fn symbol(&self) -> &'static str {
+		"LocalUSDC"
+	}
+
+	fn id(&self) -> CurrencyId {
+		CurrencyId::LocalAsset(LOCAL_ASSET_ID)
+	}
+
+	fn ed(&self) -> Balance {
+		10_000
+	}
+
+	fn decimals(&self) -> u32 {
+		6
+	}
+}
+
+/// The faked router address on the EVM side. Needed for the precompile to
+/// verify the origin of messages.
+///
+/// NOTE: This is NOT the real address of the
+///       router, but the one we are faking on the EVM side. Hence, it is fix
+///       coded here in the same way it is fixed code on the EVM testing router.
+pub const EVM_LP_INSTANCE: [u8; 20] = hex!("1111111111111111111111111111111111111111");
+
+/// The faked domain name the LP messages are coming from and going to.
+pub const EVM_DOMAIN_STR: &str = "TestDomain";
+
+/// The test domain ChainId for the tests.
+pub const EVM_DOMAIN_CHAIN_ID: u64 = 1;
+
+pub const EVM_DOMAIN: Domain = Domain::EVM(EVM_DOMAIN_CHAIN_ID);
+
+pub fn setup_full<T: Runtime>() -> impl EnvEvmExtension<T> {
+	setup::<T, _>(|evm| {
+		setup_currencies(evm);
+		setup_pools(evm);
+		setup_tranches(evm);
+		setup_investment_currencies(evm);
+		setup_deploy_lps(evm);
+		setup_investors(evm)
+	})
+}
+
+/// Default setup required for EVM <> CFG communication
+pub fn setup<T: Runtime, F: FnOnce(&mut <RuntimeEnv<T> as EnvEvmExtension<T>>::EvmEnv)>(
+	additional: F,
+) -> impl EnvEvmExtension<T> {
+	let mut env = RuntimeEnv::<T>::from_parachain_storage(
+		Genesis::default()
+			.add(genesis::balances::<T>(DEFAULT_BALANCE * CFG))
+			.storage(),
+	);
+	env.state_mut(|evm| {
+		evm_balances::<T>(DEFAULT_BALANCE * CFG);
```

Comment:
I think once the `genesis::balance()` has been modified to take into account the evm accounts this call is no longer needed

Reply:
The problem with `gensesis::balance()` is that we need to know the `ParaId` for the given chain in order to derive the evm based accounts. For this we are accessing the `ParaId` storage and this is with our current setup not possible by using `add` as it acts outside of an externalities provided environment.

Reply:
tl;dr: I tried it already and we need a bigger change for that ^^

Reply:
🤔 Mmm... makes sense. Thanks for trying it!